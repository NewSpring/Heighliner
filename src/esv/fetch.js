import fetch from "isomorphic-fetch";

export function connect() {
  return Promise.resolve(!!process.env.ESV_KEY);
}

export class ESVFetchConnector {
  baseUrl = "http://www.esvapi.org/v2/rest/passageQuery";
  key = process.env.ESV_KEY;
  count = 0;

  getFromAPI(query) {
    const label = `ESVFetchConnector${this.getCount()}`;

    const request = this.getRequest(query);

    const headers = {
      "user-agent": "Heighliner",
      "Content-Type": "application/text"
    };

    const options = { method: "GET", headers };

    console.time(label);

    return fetch(request, options)
      .then(x => x.text())
      .then(x => {
        console.timeEnd(label);
        return x;
      });
  }

  getRequest(query) {
    let request = `${this.baseUrl}?key=${this.key}`;
    request += `&passage=${query}`;
    request += "&include-headings=false";
    request += "&include-passage-references=false";
    request += "&include-footnotes=false";
    request += "&include-audio-link=false";
    request += "&include-short-copyright=false";
    return request;
  }

  getCount() {
    this.count++;
    return this.count;
  }
}
