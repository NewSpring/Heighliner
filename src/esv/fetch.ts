
import fetch from "isomorphic-fetch";

export function connect(): Promise<boolean> {
  return new Promise((cb) => {
    const hasKey = process.env.ESV_KEY;
    cb(hasKey);
  });
}

export class ESVFetchConnector {
  private baseUrl: string = "http://www.esvapi.org/v2/rest/passageQuery";
  private key: string = process.env.ESV_KEY;
  private count: number = 0;

  public get(query: string): Promise<any> {
    const label = `ESVFetchConnector${this.getCount()}`;

    const request = this.getRequest(query);

    const headers = {
      "user-agent": "Heighliner",
      "Content-Type": "application/text",
    } as { [index: string]: string };

    const options = { method: "GET", headers };

    console.time(label); // tslint:disable-line
    // XXX we CAN cache the ESV responses!

    return fetch(request, options)
      .then(x => x.text())
      .then(x => { console.timeEnd(label); return x; }); // tslint:disable-line
  }

  private getRequest(query: string): string {
    let request = `${this.baseUrl}?key=${this.key}`;
    request += `&passage=${query}`;
    request += "&include-headings=false";
    request += "&include-passage-references=false";
    request += "&include-footnotes=false";
    request += "&include-audio-link=false";
    request += "&include-short-copyright=false";
    return request;
  }

  private getCount(): number {
    this.count++;
    return this.count;
  }

}
