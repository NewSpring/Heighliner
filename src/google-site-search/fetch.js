import fetch from "isomorphic-fetch";
// import DataLoader from "dataloader";

export const connect = address => new Promise(cb => {
  cb(true);
});

export class GoogleConnector {
  count = 0;

  get(endpoint) {
    const label = `GoogleConnector${this.getCount()}`;
    const headers = {
      "user-agent": "Heighliner",
      "Content-Type": "application/json",
    };

    const options = { method: "GET", headers };
    console.time(label); // tslint:disable-line
    // XXX we can't cache google site search legally
    return fetch(endpoint, options)
      .then(x => x.json ? x.json() : x.text())
      .then(x => {
        console.timeEnd(label);
        return x;
      }); // tslint:disable-line
  }

  getCount() {
    this.count++;
    return this.count;
  }
}
