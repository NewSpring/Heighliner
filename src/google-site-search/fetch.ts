
import fetch from "isomorphic-fetch";
// import DataLoader from "dataloader";

export function connect(address: string): Promise<boolean> {
  return new Promise((cb) => {
    cb(true);
  });
}

export class GoogleConnector {
  private count: number = 0;

  public get(endpoint: string): Promise<any> {
    const label = `GoogleConnector${this.getCount()}`;
    const headers = {
      "user-agent": "Heighliner",
      "Content-Type": "application/json",
    } as { [index: string]: string };

    const options = { method: "GET", headers };
    console.time(label); // tslint:disable-line
    // XXX we can't cache google site search legally
    return fetch(endpoint, options)
      .then(x => x.json ? x.json() : x.text())
      .then(x => { console.timeEnd(label); return x; });  // tslint:disable-line
  }



  private getCount(): number {
    this.count++;
    return this.count;
  }

}
