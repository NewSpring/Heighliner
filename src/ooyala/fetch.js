export function connect() {
  return Promise.resolve(!!process.env.OOYALA_KEY);
}

