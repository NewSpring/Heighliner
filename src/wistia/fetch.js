import fetch from "isomorphic-fetch";
import { stringify } from "qs";

const BASE = "https://api.wistia.com/v1";
const UPLOAD = "https://upload.wistia.com";

export function connect() {
  return Promise.resolve(!!process.env.WISTIA_KEY);
}

export class WistiaFetchConnector {
  get(route, body) {
    return this.fetch("GET", `${BASE}/${route}.json?api_password=${process.env.WISTIA_KEY}`, JSON.stringify(body));
  }

  post(route, body) {
    return this.fetch("POST", `${BASE}/${route}.json?api_password=${process.env.WISTIA_KEY}`, JSON.stringify(body));
  }

  upload(body) {
    return this.fetch("POST", `${UPLOAD}/?api_password=${process.env.WISTIA_KEY}`, stringify(body), { "Content-Type": "application/x-www-form-urlencoded" });
  }

  fetch(method, url, body = {}, headers = {}) {
    return fetch(url, {
      headers, method, body,
    })
      .then((response) => {
        const { status, statusText, error } = response;

        if (status === 204) return { json: () => ({ status: 204, statusText: "success" }) };
        if (status >= 200 && status < 300) return response;
        if (status >= 400) {
          const err = new Error(statusText);
          err.code = status;
          throw err;
        }

        return {
          json: () => ({ status, statusText, error }),
        };
      })
      .then(x => x.json());
  }
}
