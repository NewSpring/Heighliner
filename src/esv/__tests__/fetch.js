/* eslint-disable */

import casual from "casual";

import { ESVFetchConnector, connect } from "../fetch";

describe("ESV", () => {
  it("`connect` should fail without any env vars", async () => {
    delete process.env.ESV_KEY;

    const ESV = await connect();
    expect(ESV).toBeFalsy();
  });

  it("`connect` should be fine with esv key", async () => {
    process.env.ESV_KEY = casual.word;

    const ESV = await connect();
    expect(ESV).toBeTruthy();
  });

  it("`ESVFetchConnector` should export getFromAPI function", () => {
    const testFetcher = new ESVFetchConnector();
    expect(testFetcher.getFromAPI).toBeTruthy();
  });

  it("`ESVFetchConnector` should format the query", () => {
    process.env.ESV_KEY = "test-key";

    const testFetcher = new ESVFetchConnector();
    expect(testFetcher.getRequest("john 3:16")).toMatchSnapshot();
  });

  it("`ESVFetchConnector` should keep track of connectors", () => {
    const testFetcher = new ESVFetchConnector();
    expect(testFetcher.getCount()).toEqual(1);
    expect(testFetcher.getCount()).toEqual(2);
  });
});
