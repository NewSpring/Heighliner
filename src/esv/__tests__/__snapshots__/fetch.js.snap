exports[`ESV \`ESVFetchConnector\` should format the query 1`] = `"http://www.esvapi.org/v2/rest/passageQuery?key=test-key&passage=john 3:16&include-headings=false&include-passage-references=false&include-footnotes=false&include-audio-link=false&include-short-copyright=false"`;
